File: UVA/10114.c
#include <stdio.h>

int main(){
  int months,drecords,mdepre,m;
  float dpayment, loan, carw, depre,pdepre;
  scanf("%i %f %f %i",&months,&dpayment,&loan,&drecords);
  while(months>0){
    scanf("%i %f",&m, &pdepre);
    carw = (dpayment+loan)*(1-pdepre);
    m++;
    printf("car worth:%f       loan:%f\n",carw,loan);
    for(drecords--;drecords>0;drecords--){
      scanf("%i %f",&mdepre,&depre);
      while(mdepre>m && loan>=carw){
        m++;
        carw *= 1-pdepre;
        loan -= dpayment;
        printf("car worth:%f       loan:%f\n",carw,loan);
      }
      if(mdepre == m){
        pdepre = depre;
      }
    }
    while(loan >= carw && m<=months){
      
      carw *= 1-pdepre;
      loan -= dpayment;
      m++;
      printf("--car worth:%f       loan:%f\n",carw,loan);
    }
    printf(m==1?"%i month\n":"%i months\n",m-1);
    scanf("%i %f %f %i",&months,&dpayment,&loan,&drecords);
  }
  return 0;
}

File: UVA/10114.py
l = input().split()
m,dp,loan,r = [float(e) for e in l]
while m>=0:
    rl = []
    for e in range(int(r)):
        vls =[float(i) for i in input().split()]
        vls[1] = 1-vls[1]
        rl.append(vls)
    dep = rl[0][1]
    cv = (loan+dp)*dep #car worth in month 0
    months = 0
    payment = loan/m
    i = 1
    while(loan-cv>=1e-8 and months<=m):
        months+=1
        if(i<r):
            if(rl[i][0]==months):
                dep = rl[i][1]
                i+=1
        loan-=payment
        cv *=dep
    print(months, "month" if months == 1 else "months")
    m,dp,loan,r = [float(e) for e in input().split()]